dev/src/index.ts

```diff
@@ -1186,6 +1186,39 @@ export class Firestore implements firestore.Firestore {
     this.bulkWritersCount -= 1;
   }
 
+  /**
+   * Retrieves all descendant documents nested under the provided reference.
+   *
+   * @private
+   */
+  // TODO(chenbrian): Make this a private method after adding recursive delete.
+  _getAllDescendants(
+    ref: CollectionReference | DocumentReference
+  ): Promise<DocumentReference[]> {
+    const firestore =
+      ref instanceof CollectionReference
+        ? ref._firestore
+        : ref.parent._firestore;
+    const segments = ref.path.split('/');
```

Comment:
Can you make CollectionReference and DocumentReference implement the same ResourcePath-exposing property? Then you can access the path as ref._resourcePath or something and don't need to split.

Reply:
Done.